### PR TITLE
Add release for android OS

### PR DESCRIPTION
**Pull request by uudashr**
Add release for android. Some of use Termux on Android as alternative development machine.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -44,7 +44,6 @@ linters-settings:
       - dupImport # https://github.com/go-critic/go-critic/issues/845
       - ifElseChain
       - octalLiteral
-      - whyNoLint
```
Comment (tpounds) [inline .golangci.yml]: Unfortunately, this is needed for latest v1.22.x releases due to go-critic update. The GitHub check should pass once https://github.com/go-critic/go-critic/issues/845 is merged and deployed.

Comment (uudashr) [reply .golangci.yml]: Any suggestion what should I do now?
1. Return the `whyNoLint` config
2. ???

Comment (tpounds) [reply .golangci.yml]: Just undo the `whyNoLint` removal commit and it should be good to merge.

Comment (ernado) [reply .golangci.yml]: GolangCI failure is expected, yes.

To undo this commit you can do the following:
```
git reset --hard 6fd38300fbdc897e6786e899717e712cbecde0b8
git push -f uudashr 
```
Where 6fd38300fbdc897e6786e899717e712cbecde0b8 is your first commit, and `uudashr` is your fork remote name.

Comment (uudashr) [reply .golangci.yml]: Reverted

Comment (uudashr) [reply .golangci.yml]: Hi @tpounds already undo the `whyNoLint`, should good now

**Comment by jirfag** [comment]
thank you!

**Comment by jirfag** [comment]
I've needed to revert this change because a new release can't be built on android. I got errors like these:
```
   ⨯ release failed after 35.30s error=failed to build for android_amd64: # github.com/golangci/golangci-lint/cmd/golangci-lint
/usr/local/go/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
ld: warning: option -s is obsolete and being ignored
ld: unknown option: -z
clang: error: linker command failed with exit code 1 (use -v to see invocation)

   ⨯ release failed after 621.89s error=failed to build for android_arm64: # github.com/golangci/golangci-lint/cmd/golangci-lint
/usr/local/go/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
clang: error: invalid linker name in argument '-fuse-ld=gold'
```

**Comment by tpounds** [comment]
@jirfag AFAICT, we need to have `cgo` and the Android NDK to build this release target.

**Comment by jirfag** [comment]
thank you,
@tpounds @uudashr will you try to make a new pull request with adding needed libraries? If yes - please check locally that `make deploy` works (set flags to just build but not deploy).

**Comment by uudashr** [comment]
Somehow go get also failed on android

```
$ GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@master
go: finding github.com/golangci/golangci-lint master
go: finding github.com/golangci master
go: finding github.com/golangci/golangci-lint/cmd/golangci-lint master
go: finding github.com/golangci/golangci-lint/cmd master
go: finding github.com master
go: downloading github.com/golangci/golangci-lint v1.23.6
go: extracting github.com/golangci/golangci-lint v1.23.6
# github.com/golangci/golangci-lint/cmd/golangci-lint
/data/data/com.termux/files/usr/lib/go/pkg/tool/android_arm64/link: running aarch64-linux-android-clang failed: exit status 1
clang-9: error: invalid linker name in argument '-fuse-ld=gold'
```

**Pull request merged**